### PR TITLE
Move to closest path position when not on path

### PR DESCRIPTION
**Pull request by Somotaw**

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
DCBOT: 
This repository is under [democratic collaboration](https://github.com/TooAngel/democratic-collaboration) and will be merged automatically.

The merge decision is based on the outcome of the reviews:
 - `Approve` add the reviewer value (number of commits) to the `metric`
 - `Request changes` substract the reviewer value from the `metric`

 The merge will happen after `(1 - metric/total.votes) * (5 + pull_request.commits * 5))` calculated from the last code change.

Please review the PR to make a good democratic decision.

Summoning some reviewers:
 - @TooAngel: I always like your reviews
 - @Nevrdid: Not sure why you are here
 - @WhiteHalmos: Please look at the code

**Comment by democratic-collaboration** [comment]
DCBOT: Adding WIP label, the title is prefixed with [WIP]

**Comment by Somotaw** [comment]
better? :)

File: src/prototype_creep_routing.js [hunk 1]
```diff
@@ -309,8 +309,21 @@ Creep.prototype.moveByPathMy = function(route, routePos, start, target, skipPreM
...
+    //let closestPathRange = 50;
+    //let closestPathPosition;
+    //for (let iterator = 0; iterator < search.path.length; iterator++) {
+    //  if (this.pos.getRangeTo(search.path[iterator] < closestPathRange)) {
+    //    closestPathPosition = iterator;
+    //    closestPathRange = this.pos.getRangeTo(search.path[iterator] < closestPathRange);
+    //  }
+    //}
+
+    let returnCode = this.moveTo(_.min(search.path, function(object) {
```
Comment (TooAngel) [inline src/prototype_creep_routing.js]: It doesn't matter, but I tried to pass always the position to method like this. Also with the background to overwrite the method and only have to handle positions.

Much better.

Comment (Somotaw) [inline src/prototype_creep_routing.js]: you mean this.pos instead of this?

Comment (TooAngel) [inline src/prototype_creep_routing.js]: Exactly

**Comment by democratic-collaboration** [comment]
DCBOT: Removing WIP label, the title is not prefixed with [WIP]

**Comment by democratic-collaboration** [comment]
DCBOT: A new review, yeah.

    Votes: 182/302
    Coefficient: 0.602649006623
    Merging in 3 days 16 hours
    Age 0 days 7 hours

**Comment by democratic-collaboration** [comment]
DCBOT: A new review, yeah.

    Votes: 189/302
    Coefficient: 0.62582781457
    Merging in 3 days 9 hours
    Age 0 days 8 hours

**Comment by democratic-collaboration** [comment]
DCBOT: A new review, yeah.

    Votes: 238/302
    Coefficient: 0.788079470199
    Merging in 2 days -8 hours
    Age 0 days 10 hours

**Comment by TooAngel** [comment]
Resolving the conflicts would be nice

**Pull request merged**